Orabot.Core/WatcherServices/ResourceCenterMapWatcherService.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Discord.WebSocket;
+using Microsoft.Extensions.Configuration;
+using Orabot.Core.Abstractions;
+using Orabot.Core.Integrations.ResourceCenter;
+using Orabot.Core.Objects.OpenRaResourceCenter;
+using Orabot.Core.Transformers.LinkToEmbedTransformers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orabot.Core.WatcherServices
+{
+	internal class ResourceCenterMapWatcherService : ILongRunningService
+	{
+		private readonly string ServerName;
+		private readonly string ChannelName;
+
+		private readonly DiscordSocketClient _discordClient;
+		private readonly IMapsApi _mapsApi;
+		private readonly OpenRaResourceCenterMapLinkToEmbedTransformer _toEmbedTransformer;
+
+		private int lastKnownMapId = -1;
+
+		public ResourceCenterMapWatcherService(
+			DiscordSocketClient discordClient,
+			IConfiguration configuration,
+			IMapsApi mapsApi,
+			OpenRaResourceCenterMapLinkToEmbedTransformer toEmbedTransformer)
+		{
+			_discordClient = discordClient;
+			_mapsApi = mapsApi;
+			_toEmbedTransformer = toEmbedTransformer;
+
+			var mapAnnouncer = configuration.GetRequiredSection("MapAnnouncer");
+			ServerName = mapAnnouncer["ServerName"];
+			ChannelName = mapAnnouncer["ChannelName"];
+		}
+
+		public async Task ExecuteAsync(CancellationToken cancellationToken)
+		{
+			while (!cancellationToken.IsCancellationRequested)
+			{
+				try
+				{
+					Console.WriteLine($"Checking resource center for new maps... {DateTime.Now}");
+
+					var response = await _mapsApi.GetMaps();
+					var maps = response.Values;
+					if (lastKnownMapId == -1)
+						lastKnownMapId = maps.First().Id;
+					else
+						await HandleNewMaps(maps);
+				}
+				catch (Exception ex)
+				{
+					Console.WriteLine(ex.ToString());
+				}
+
+				// Pause for 10 minutes between checks.
+				await Task.Delay(600_000);
+			}
+		}
+
+		private async Task HandleNewMaps(IEnumerable<MapInfo> maps)
+		{
+			var guild = _discordClient.Guilds.FirstOrDefault(x => x.Name == ServerName);
+			var channel = guild?.Channels?.FirstOrDefault(x => x.Name == ChannelName);
```

Comment:
Neither of those depends on `maps`, so could/should be done in the ctor?

Reply:
We can't do them in the constructor because at that point the discordClient hasn't connected yet. But we could probably make them lazy 🤷‍♂️

Reply:
Hm, ok.